### DISCUSSION TITLE
XML_C14N_1_0, and XML_C14N_1_1 Canonicalization

### DISCUSSION THREAD

**Discussion by apokalipto** [sparklemotion/nokogiri#3150] [General Q&A] [answered]
Hi, I came accross an issue in versions > 1.14, where XML_C14N_1_0, and XML_C14N_1_1 are not supported (https://github.com/sparklemotion/nokogiri/blob/f2563b79dea8c5ee04c3a61491650a3e42853ec8/ext/nokogiri/xml_document.c#L628)

Why are these methods not supported in newer versions ? Is there a way around it?

**Comment by flavorjones** [comment, marked as answer]
That line of code was introduced in 08823e01 which contains a descriptive commit message that I'll reproduce here to save you a click:

```
commit 08823e01
Author: Mike Dalessio <mike.dalessio@gmail.com>
Date:   2022-01-19 23:39:02 -0500

    fix: Document#canonicalize raises exception for incompatible modes

    `Document#canonicalize` now raises an exception if
    `inclusive_namespaces` is non-nil and the mode is inclusive,
    i.e. XML_C14N_1_0 or XML_C14N_1_1.

    `inclusive_namespaces` can only be passed with exclusive modes, and
    previously this silently failed.
```

You should only see this error if you're trying to do something that the canonicalization spec doesn't support. If you think this is incorrect behavior, can you show your code and your input XML?

**Reply by apokalipto** [reply]
Thank you very very much! This was exactly my mistake! I was passing the array of inclusive_namespaces.
